### PR TITLE
dotCMS/core#18294 Create drop zone web component

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/models/dot-http-error-file-response.model.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface DotHttpErrorFileResponse {
```
Comment (hmoreras) [inline src/models/dot-http-error-file-response.model.ts:1]: need to move this to dotcms-models

File: src/models/dot-http-request-options.model.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface DotHttpRequestOptions {
```
Comment (hmoreras) [inline src/models/dot-http-request-options.model.ts:1]: need to move this to dotcms-models

File: src/components/contenttypes-fields/dot-form/services/dot-upload.service.ts [hunk 3]
```diff
@@ -44,27 +51,74 @@ export class DotUploadService {
...
-            method: 'POST',
-            headers: {
-                Origin: window.location.hostname
+        if (Array.isArray(data)) {
+            data.forEach((file: File) => {
+                formData.append('files', file);
+            });
+        } else {
+            formData.append('file', data);
+        }
```
Comment (fmontes) [inline src/components/contenttypes-fields/dot-form/services/dot-upload.service.ts:78]:
```suggestion
        const files = Array.isArray(data) ? data | [data]]
        files.forEach((file: File) => {
                formData.append('files', file);
        });
```

File: src/components/contenttypes-fields/dot-form/services/dot-upload.service.ts [hunk 4]
```diff
@@ -44,27 +51,74 @@ export class DotUploadService {
...
+            xhr.open(opts.method || 'get', url);
+            for (let name in opts.headers || {}) {
+                xhr.setRequestHeader(name, opts.headers[name]);
+            }
+            xhr.onload = () => res(xhr);
+            xhr.onerror = rej;
+            if (xhr.upload && progressCallBack) {
+                xhr.upload.onprogress = (e: ProgressEvent) => {
+                    const percentComplete = e.loaded / e.total * 100;
+                    progressCallBack(percentComplete);
```
Comment (fmontes) [inline src/components/contenttypes-fields/dot-form/services/dot-upload.service.ts:118]: Isn't better just to send the `e: ProgressEvent` to the `progressCallBack`?

File: src/elements/dot-progress-bar/dot-progress-bar.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        &:before {
+            content: "";
+            background-color: var(--color-sec);
+            width: calc(var(--progress-width, 0) * 1%);
+            display: flex;
+            position: absolute;
+            left: 0px;
+            bottom: 0px;
+            top: 0px;
+            max-width: 100%;
```
Comment (fmontes) [inline src/elements/dot-progress-bar/dot-progress-bar.scss:22]:
```suggestion
            max-width: 100%;
            transition: width 100ms;
```

**Pull request merged**